src/core/thread/network_diagnostic.cpp

```diff
@@ -141,17 +141,18 @@ void NetworkDiagnostic::HandleDiagnosticGetResponse(Coap::Message *         aMes
                                                     const Ip6::MessageInfo *aMessageInfo,
                                                     otError                 aResult)
 {
-    VerifyOrExit(aResult == OT_ERROR_NONE, OT_NOOP);
-    VerifyOrExit(aMessage && aMessage->GetCode() == Coap::kCodeChanged, OT_NOOP);
-
-    otLogInfoNetDiag("Received diagnostic get response");
+    SuccessOrExit(aResult);
+    VerifyOrExit(aMessage && aMessage->GetCode() == Coap::kCodeChanged, aResult = OT_ERROR_FAILED);
```

Comment:
```suggestion
    VerifyOrExit(aMessage->GetCode() == Coap::kCodeChanged, aResult = OT_ERROR_FAILED);
```
aMessage must not be nullptr if success?

Reply:
@wgtdkp , thoughts on this?